portal-kernel/src/com/liferay/exportimport/kernel/lar/StagedModelDataHandler.java

```diff
@@ -73,23 +158,91 @@ public void importCompanyStagedModel(
 			PortletDataContext portletDataContext, String uuid, long classPK)
 		throws PortletDataException;
 
+	/**
+	 * "Imports" missing reference. When a reference is exported as missing, the
+	 * framework calls this method during the import process. This method
+	 * updates the new primary key map in <code>portletDataContext</code>. In
+	 * other words it maps the ID of the existing staged model to the old ID
+	 * which is in the <code>referenceElement</code>.
+	 *
+	 * @param  portletDataContext the portlet data context of the current
+	 *         process
+	 * @param  referenceElement XML element that contains information about the
+	 *         reference
+	 * @throws PortletDataException
+	 */
 	public void importMissingReference(
 			PortletDataContext portletDataContext, Element referenceElement)
 		throws PortletDataException;
 
+	/**
+	 * When this <code>StagedModelDataHandler</code> extends {@link
+	 * BaseStagedModelDataHandler}, {@link
+	 * BaseStagedModelDataHandler#importMissingReference(PortletDataContext,
+	 * Element)} and {@link
+	 * BaseStagedModelDataHandler#doImportMissingReference(PortletDataContext,
+	 * Element)} are not overridden, this method is called.
```

Comment:
@codyhoag the importMissingReference is a little bit misleading name.

When we export something as a missing reference, we sort expect that those will be available on the import side but we need to look for them somehow and validate if they are there.
The purpose of the importMissingReference is actually not importing a missing reference, but to try to locate and map the missing reference to the information we have from the export side.
So if we have a FileEntry as a missing reference we have an ID for it on the export side let's say 2. We have that information from the LAR but not the FileEntry itself. When we are importing we look for the FileEntry but it might have a different ID let's say 4. We need to map those to together for the process for later usage.

The importMissingReference method does that. Tries to locate the missing reference and actually not importing anything, but mapping the correct value to the value we have from the export side.